public/js/clients/firefox.js

```diff
@@ -40,16 +42,35 @@ function presentTabs(tabs) {
   });
 }
 
-function connectClient(onConnect) {
+function connectClient() {
+  const deferred = defer();
+  let isConnected = false;
+
   const socket = new WebSocket("ws://localhost:9000");
   const transport = new DebuggerTransport(socket);
   currentClient = new DebuggerClient(transport);
 
+
+  setTimeout(() => {
+    if (isConnected) {
+      return;
+    }
+
+    deferred.resolve([]);
+  }, 1000);
```

Comment:
Nice, I like that this is closer to the connection code and there's not a top-level `setTimeout` any more.